src/app/zap-templates/templates/app/attribute-size-src.zapt

```diff
@@ -0,0 +1,134 @@
+{{> header}}
+
+ #include <app/util/af.h>
+ #include <app/util/basic-types.h>
+ #include <support/SafeInt.h>
+ #include <support/logging/CHIPLogging.h>
+
+ #include "gen/af-structs.h"
+
+ using namespace chip;
+
+ // The first 2 bytes specify the number of entries. A value of 0xFFFF means the list in invalid
+ // and data is undefined.
+ constexpr uint16_t kSizeLengthInBytes = 2u;
+
+ void copyListMember(uint8_t * dest, uint8_t * src, bool write, uint16_t * offset, uint16_t length)
+ {
+     if (write)
+     {
+         memmove(dest + *offset, src, length);
+     }
+     else
+     {
+         memmove(dest, src + *offset, length);
+     }
+
+     *offset = static_cast<uint16_t>(*offset + length);
+ }
+
+ uint16_t emberAfCopyList(ClusterId clusterId, EmberAfAttributeMetadata * am, bool write, uint8_t * dest, uint8_t * src, int32_t index)
+ {
+     if (index == -1)
+     {
+         memmove(dest, src, am->size);
+         return am->size;
+     }
+
+     if (index == 0)
+     {
+         memmove(dest, src, kSizeLengthInBytes);
```

Comment:
So this assumes length is stored native-endian.  But `emberAfAttributeValueListSize` below uses `emberAfGetInt16u` which assumes little-endian.  Which of these callsites is wrong?

Reply:
Thanks for catching this!